### PR TITLE
Define JACK headers and sources globally and add some documentation 

### PR DESCRIPTION
**Pull request by ann0see**
**Short description of changes**
Instead of using hardcoded paths to the sound files @pgScorpio suggested to use variables. This PR uses global definitions for JACK sources and headers and adds some small comments.

CHANGELOG: Internal: Improved maintainability of Jamulus.pro by using global definitions.

**Context: Fixes an issue?**
Fixes: #2573

**Does this change need documentation? What needs to be documented and how?**
No

**Status of this Pull Request**
Ready for testing

**What is missing until this pull request can be merged?**
Needs whole CI to pass

## Checklist

-  [x] I've verified that this Pull Request follows the [general code principles](https://github.com/jamulussoftware/jamulus/blob/main/CONTRIBUTING.md#jamulus-projectsource-code-general-principles)
-  [x] I tested my code and it does what I want (Linux (JACK gives no warning and sound is OK), Windows too. I can't test macOS at the moment - but build is OK.)
-  [x] My code follows the [style guide](https://github.com/jamulussoftware/jamulus/blob/main/CONTRIBUTING.md#source-code-consistency) 
-  [x] I waited some time after this Pull Request was opened and all GitHub checks completed without errors. 
-  [x] I've filled all the content above

AUTOBUILD: Please build all targets

### CODE REVIEW

File: Jamulus.pro [hunk 1]
```diff
@@ -68,6 +68,35 @@ INCLUDEPATH_OPUS = libs/opus/include \
     libs/opus/silk/fixed \
     libs/opus
 
+# Add include paths and headers for new audio APIs here
+
+# ASIO
```
Comment (pljones) [inline Jamulus.pro:73]: I'd split this into two parts:
- provided by the SDK
- "part of Jamulus"

Comment (pljones) [reply Jamulus.pro:73]: I'd also make _all_ the variable definitions be either
```
VARNAME =
```
or
```
VARNAME = value
```
depending on the config that makes them relevant.

**Comment by ann0see** [comment]
Hmm. Then I misunderstood you. I thought you were referring to splitting the ASIO part out to another variable? Can you please provide a suggestion what you'd like to see?

File: Jamulus.pro [hunk 2]
```diff
@@ -68,6 +68,34 @@ INCLUDEPATH_OPUS = libs/opus/include \
...
+SOURCES_COREAUDIO_MAC = src/sound/coreaudio-mac/sound.cpp
+
+# CoreAudio iOS
+HEADERS_COREAUDIO_IOS = src/sound/coreaudio-ios/sound.h
+SOURCES_COREAUDIO_IOS = src/sound/coreaudio-ios/sound.mm
+
+# Oboe
+HEADERS_OBOE = src/sound/oboe/sound.h
+SOURCES_OBOE = src/sound/oboe/sound.cpp \
+        src/android/androiddebug.cpp # TODO: Remove debugging code if no longer needed
```
Comment (pljones) [inline Jamulus.pro:97]: Should line 97 be done more like
```
SOURCES_OBOE = src/sound/oboe/sound.cpp
# TODO: Remove debugging code if no longer needed else move to src/sound if part of sound
SOURCES_OBOE += src/android/androiddebug.cpp
```

**Comment by pljones** [comment]
I'm starting to think this is creating a lot of variables for no benefit.

The variables should only be added to the overall HEADERS, etc, variables conditionally - so wouldn't it be better to do _that_ at the time these are being set?

File: Jamulus.pro (hunk 2 above)
Comment (ann0see) [reply Jamulus.pro]: Ok. Yes. Done.

**Comment by ann0see** [comment]
@pljones so you mean basically that we should revert this? I think @pgScorpio mainly wanted to get rid of the duplication introduced by the JACK headers.

**Comment by pljones** [comment]
> @pljones so you mean basically that we should revert this? I think @pgScorpio mainly wanted to get rid of the duplication introduced by the JACK headers.

Are the variables used in multiple places?  Otherwise there's no duplication being removed.

**Comment by ann0see** [comment]
The JACK headers are used for all three main OS.

**Comment by ann0see** [comment]
@pljones I've now updated my PR and split it into two commits. I hope that's ok for all.

**Review by pljones — APPROVED** [review]

**Pull request merged**